### PR TITLE
fix the 1756, add extra test for nested bug(callback not being invoked)

### PR DESCRIPTION
**Pull request by MacondoExpress**
# Description

Fix PR for the bug #1756, during the fix it was found that the callback fields where not getting resolved inside the onCreate statement.

# Issue

Closes:

- #1756 

# Checklist

The following requirements should have been met (depending on the changes in the branch):

- [x] TCK tests have been updated
- [x] Integration tests have been updated
- [x] New files have copyright header
- [x] CLA (https://neo4j.com/developer/cla/) has been signed

### CODE REVIEW

File: packages/graphql/src/schema/to-compose.ts [hunk 1]
```diff
@@ -86,7 +86,7 @@ export function objectFieldsToComposeFields(fields: BaseField[]): {
 
 export function objectFieldsToCreateInputFields(fields: BaseField[]): Record<string, InputField> {
     return fields
-        .filter((f) => !(f as PrimitiveField)?.autogenerate && !(f as TemporalField)?.timestamps)
+        .filter((f) => !(f as PrimitiveField)?.autogenerate && !(f as PrimitiveField)?.callback && !(f as TemporalField)?.timestamps)
```
Comment (angrykoala) [inline packages/graphql/src/schema/to-compose.ts:89]: Maybe it is worth using intermediate variables to split the complexity of this boolean logic?

Something akin to:
```suggestion
        .filter((f) => {
        const isAutogenerate=(f as PrimitiveField)?.autogenerate
        const isCallback=(f as PrimitiveField)?.autogenerate
        const isTemporal=(f as TemporalField)?.timestamps
       
        return !isAutogenerate && !isCallback && !isTemporal
        });
```

It is more verbose, but I think it helps a lot in readability

File: packages/graphql/tests/integration/issues/1756.int.test.ts [hunk 2]
```diff
@@ -0,0 +1,207 @@
...
+    const genreType = generateUniqueType("Genre");
+
+    let schema: GraphQLSchema;
+    let driver: Driver;
+    let neo4j: Neo4j;
+
+    beforeAll(async () => {
+        neo4j = new Neo4j();
+        driver = await neo4j.getDriver();
+        // user bug
```
Comment (angrykoala) [inline packages/graphql/tests/integration/issues/1756.int.test.ts:37]: This comment seems a bit irrelevant

File: packages/graphql/tests/integration/issues/1756.int.test.ts [hunk 3]
```diff
@@ -0,0 +1,207 @@
...
+        expect(result.errors).toBeFalsy();
+        expect(result?.data?.[productType.operations.create]).toEqual({
+            [productType.plural]: [{
+                id: "callback_value"
+            }]
+        });
+    });
+});
+
+describe("https://github.com/neo4j/graphql/issues/1756, nested bug", () => {
```
Comment (angrykoala) [inline packages/graphql/tests/integration/issues/1756.int.test.ts:112]: :thinking: It seems that this shares the type definitions, maybe it makes sense to have this as a new test case within the same `describe` instead of a separate one?

**Review by angrykoala — APPROVED** [review]
Looks good, just a couple of proposals in the code and tests

**Comment by neo4j-team-graphql** [comment]
Made some changes to the GraphQL Toolbox? Or made some changes to the core library perhaps?

Check out your changes at
 http://neo4j-graphql-toolbox-stage-1842.surge.sh (=> for local development; use for instance [Neo4j Desktop](https://neo4j.com/download/))
or
 https://neo4j-graphql-toolbox-stage-1842.surge.sh (=> use to connect to an [Aura Free](https://neo4j.com/cloud/aura-free/) dbms)

**Review by darrellwarde — CHANGES_REQUESTED** [review]
Looks good, just missing a changeset!!! 🙂

**Comment by MacondoExpress** [comment]
> Looks good, just missing a changeset!!! 🙂

Oh no! I'll add it :)

**Review by darrellwarde — APPROVED** [review]
